controllers/azurestackedgemachine_controller.go

```diff
@@ -359,49 +359,49 @@ func (r *AzureMachineReconciler) reconcileNormal(ctx context.Context, machineSco
         return reconcile.Result{}, errors.Wrapf(err, "VM deleted, retry creating in next reconcile")
     default:
         machineScope.V(2).Info("VM state is undefined", "id", machineScope.GetVMID())
-        conditions.MarkUnknown(machineScope.AzureMachine, infrav1.VMRunningCondition, "", "")
+        conditions.MarkUnknown(machineScope.AzureStackEdgeMachine, infrav1.VMRunningCondition, "", "")
         machineScope.SetNotReady()
     }
 
     return reconcile.Result{}, nil
 }
 
-func (r *AzureMachineReconciler) reconcileAzureMachineIPAddress(machineScope *scope.MachineScope, ctx context.Context, clusterScope *scope.ClusterScope, azureIPPoolScope *scope.AzureIPPoolScope) (reterr error) {
+func (r *AzureStackEdgeMachineReconciler) reconcileAzureStackEdgeMachineIPAddress(machineScope *scope.MachineScope, ctx context.Context, clusterScope *scope.ClusterScope, azureStackEdgeIPPoolScope *scope.AzureStackEdgeIPPoolScope) (reterr error) {
     machineScope.Info("Reconciling IP Address")
     machineIP := ""
     if machineScope.IsControlPlane() {
         machineIP = clusterScope.APIServerIP()
     }
 
-	err := azureIPPoolScope.ReconcileIPs(ctx, machineScope, "ase-ip-pool", machineIP)
+	err := azureStackEdgeIPPoolScope.ReconcileIPs(ctx, machineScope, "ase-ip-pool", machineIP)
```

Comment:
Why is this reconcile is on the azureStackEgdeIPPoolScope and not on service, similar to vnet and resource groups?

Reply:
For any azure resources where we call into the Azure GO sdk to reconcile the resources, we create service for them and call into the respective compute, network client APIs. Here we are retrieving and modifying the kubernetes IPPool custom resource that we have created. This is similar to cluster and machine custom resource objects. This is done under cloud/scope/cluster.go and cloud/scope/machine.go. Hence kept the structure same to align with other custom resources. All the logic to retrieve, modify the IP pool custom resource is placed under cloud/scope/azurestakedgeippool.go